#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>
#include<QMouseEvent>
#include"mystorage.h"
#include"shapes.h"

QT_BEGIN_NAMESPACE
namespace Ui {
class MainWindow;
}
QT_END_NAMESPACE

class MainWindow : public QMainWindow
{
    Q_OBJECT

public:
    MainWindow(QWidget *parent = nullptr);
    ~MainWindow();

private:
    Ui::MainWindow *ui;
    void mousePressEvent(QMouseEvent *event) override;
    void mouseMoveEvent(QMouseEvent *event) override;
    void mouseReleaseEvent(QMouseEvent *) override;
    void keyPressEvent(QKeyEvent *event) override;
    void EditColorByRadioBtton();
    void onShapeEditPressed(Shape *sh);
    Shape* GiveMe();
    MyStorage *store;
    QPoint b;
    QPoint e;
    QPoint delta;
    QPoint postpoint;
    QString color;
    Shape *s = nullptr;
    bool isSelecting = false;
    bool groupResizing = false;
    QPoint lastResizePos;
};
#endif // MAINWINDOW_H
